### ISSUE TITLE
Change detector calls, so that only hostname is needed in detectors configuration

### ISSUE THREAD

**Issue by mdevino** [foundation-model-stack/fms-guardrails-orchestrator#142]
## Description

As an orchestrator contributor, I want to change the code calling detector endpoints, so that I can set only the hostname in the `hostname` parameter in detectors configuration (e.g. `https://my-detector.com` instead of `https://my-detector.com/api/v1/text/generation`).

Note: This should be done for all detector endpoints.

## Discussion

This was raised as an issue as [a comment in one of the recent PRs](https://github.com/foundation-model-stack/fms-guardrails-orchestrator/pull/116#discussion_r1697668762). Making this change would allow two things:

1. Support detectors that implement more than one endpoint ([detectors API reference](https://foundation-model-stack.github.io/fms-guardrails-orchestrator/?urls.primaryName=Detector+API)).
2. 

I've tried the following:

```diff
    /// Invokes detectors implemented with the `/api/v1/text/generation` endpoint
    pub async fn generation_detection(
        &self,
        model_id: &str,
        request: GenerationDetectionRequest,
    ) -> Result<Vec<DetectionResult>, Error> {
        let client = self.client(model_id)?;
-        let url = client.base_url().as_str();
+       let url = client.base_url().join("/api/v1/text/generation").unwrap();
        let response = client
-            .post(url)
+            .post(url.as_str())
            .header(DETECTOR_ID_HEADER_NAME, model_id)
            .json(&request)
            .send()
            .await?;
        if response.status() == StatusCode::OK {
            Ok(response.json().await?)
        } else {
            let error = response.json::<DetectorError>().await.unwrap();
            Err(error.into())
        }
    }
```

As a result, detector hostnames could be configured as desired. I also noticed that, if a detector was configured with a full url (say `https://mydetector.com/a/full/url`), the path was entirely replaced (i.e. `https://mydetector.com/a/full/url` becomes `https://mydetector.com/api/v1/text/generation`, not `https://mydetector.com/a/full/url/api/v1/text/generation`.

## Acceptance Criteria

- [ ] Unit tests cover new/changed code
- [ ] Examples build against new/changed code

**Comment by mdevino** [comment]
This is going to be addressed by  #205.

**Comment by declark1** [comment]
This is required for the client refactor, so I am adding validation to ensure only base urls are set in the config for http clients.
